src/Controller/DownloadController.php

```diff
@@ -93,60 +97,71 @@ public function defaultAction(int $id)
     /**
      * Set up direct download via HTTP and produce html for direct download splash screen.
      *
-     * @param integer                      $id                           The id of the dataset to download.
-     * @param DataStore                    $dataStore                    The data store.
+     * @param Dataset                      $dataset                      The id of the dataset to download.
+     * @param Datastore                    $dataStore                    The data store.
      * @param LogActionItemEventDispatcher $logActionItemEventDispatcher The log action dispatcher.
      *
      * @Route("/download/{id}/http", name="pelagos_app_download_http")
      *
      * @return Response
      */
-    public function httpAction(int $id, DataStore $dataStore, LogActionItemEventDispatcher $logActionItemEventDispatcher)
+    public function httpAction(Dataset $dataset, Datastore $dataStore, LogActionItemEventDispatcher $logActionItemEventDispatcher)
     {
-        $dataset = $this->entityHandler->get(Dataset::class, $id);
-        $downloadFileInfo = $dataStore->getDownloadFileInfo($dataset->getUdi(), 'dataset');
-        $username = null;
-        if ($this->getUser()) {
-            $username = $this->getUser()->getUsername();
-        } else {
-            $username = '';
-        }
-        $uniqueDirectory = uniqid(
-            preg_replace('/\s/', '_', $username) . '_'
-        );
-        $downloadDirectory = $this->downloadBaseDir . '/' . $uniqueDirectory;
-        mkdir($downloadDirectory, 0755);
-        $datasetFileName  = $dataset->getDatasetSubmission()->getDatasetFileName();
-        symlink(
-            $downloadFileInfo->getRealPath(),
-            $downloadDirectory . '/' . $datasetFileName
+        $datasetSubmission = $dataset->getDatasetSubmission();
+        $response = new Response(
+            null,
+            Response::HTTP_NO_CONTENT
         );
+        if ($datasetSubmission instanceof DatasetSubmission) {
+            $fileset = $datasetSubmission->getFileset();
+
+            if ($fileset instanceof Fileset) {
+                if (!$fileset->doesZipFileExist()) {
+                    $filePhysicalPath = $fileset->getProcessedFiles()->first()->getPhysicalFilePath();
```

Comment:
Make sure this is something, I tried it with an old dataset, and it didn't work because it didn't have files.